packages/zowe-explorer/src/utils/ProfilesUtils.ts

```diff
@@ -327,9 +331,17 @@ export class ProfilesUtils {
             if (credentialManagerMap && isVSCodeCredentialPluginInstalled) {
                 return this.setupCustomCredentialManager(credentialManagerMap);
             }
+            return this.setupDefaultCredentialManager();
         }
 
-        return this.setupDefaultCredentialManager();
+        const profileInfo = new imperative.ProfileInfo("zowe", {});
+        const workspacePath = ZoweVsCodeExtension.workspaceRoot?.uri.fsPath;
+        // Trigger initialize() function of credential manager to throw an error early if failed to load
+        await profileInfo.readProfilesFromDisk({
+            homeDir: FileManagement.getZoweDir(),
+            projectDir: workspacePath ? FileManagement.getFullPath(workspacePath) : undefined,
+        });
+        return profileInfo;
```

Comment:
Would it be possible to call the `profileCache.getProfileInfo` method here?

If loading the default credential manager there is a problem, we could enhance it to use the (currently unused?) `_envTheia` property to avoid loading the default credential manager.

Reply:
I guess you want an implementation like this?

```ts
    public static async getProfileInfo(secureCredentials = true): Promise<imperative.ProfileInfo> {
        let mProfileInfo: imperative.ProfileInfo;
        if (secureCredentials) {
            mProfileInfo = new imperative.ProfileInfo("zowe", {
                // eslint-disable-next-line @typescript-eslint/no-unsafe-return
                credMgrOverride: imperative.ProfileCredentials.defaultCredMgrWithKeytar(ProfilesCache.requireKeyring),
            });
        } else {
            mProfileInfo = new imperative.ProfileInfo("zowe", {});
        }
        await mProfileInfo.readProfilesFromDisk({ homeDir: FileManagement.getZoweDir(), projectDir: this.cwd ?? undefined });
        return mProfileInfo;
    }
```

I was not able to wire this one up so it could be called from ProfilesUtils without causing a large amount of changes. The cache in the extension is accessed via Profiles, which has its own implementation of getProfileInfo() that overloads it. Can you help?

Reply:
> As Peter mentioned, changing this will cause a cascade of other changes, so maybe this effort could be done separately?

I agree with this as the change would cause a cascade of other ones, and I think the effort has already been done in https://github.com/zowe/zowe-explorer-vscode/pull/3296